src/System.Security.SecureString/src/System/Security/SecureString.Windows.cs

```diff
@@ -0,0 +1,361 @@
+        }
+
+        private void EnsureCapacity(ref SafeBSTRHandle decryptedBuffer, int capacity)
+        {
+            if (capacity > MaxLength)
+            {
+                throw new ArgumentOutOfRangeException(nameof(capacity), SR.ArgumentOutOfRange_Capacity);
+            }
+
+            Debug.Assert(capacity > _decryptedLength, $"Expected to only be called when growing capacity = {capacity}, _decryptedLength = {_decryptedLength}");
+
+            SafeBSTRHandle newBuffer = SafeBSTRHandle.Allocate((uint)capacity);
+            SafeBSTRHandle.Copy(decryptedBuffer, newBuffer, (uint)_decryptedLength * sizeof(char));
+            decryptedBuffer.Dispose();
+            decryptedBuffer = newBuffer;
+        }
+
+        private void ProtectMemory(SafeBSTRHandle decryptedBuffer)
+        {
+            Debug.Assert(!decryptedBuffer.IsInvalid, "Invalid buffer!");
+
+            if (_decryptedLength == 0)
+            {
+                return;
+            }
+
+            try
+            {
+                SafeBSTRHandle newEncryptedBuffer;
+                if (!CryptProtectData(decryptedBuffer, out newEncryptedBuffer))
+                {
+                    throw new CryptographicException(Marshal.GetLastWin32Error());
+                }
+
+                _encryptedBuffer.Dispose();
+                _encryptedBuffer = newEncryptedBuffer;
+            }
+            finally
+            {
+                decryptedBuffer.ClearBuffer();
+                decryptedBuffer.Dispose();
+            }
+        }
+
+        private SafeBSTRHandle UnprotectMemory()
+        {
+            Debug.Assert(!_encryptedBuffer.IsInvalid, "Invalid buffer!");
+
+            if (_decryptedLength == 0)
+            {
+                return _encryptedBuffer;
+            }
+
+            SafeBSTRHandle decryptedBuffer;
+            if (!CryptUnprotectData(_encryptedBuffer, out decryptedBuffer))
+            {
+                throw new CryptographicException(Marshal.GetLastWin32Error());
+            }
+
+            return decryptedBuffer;
+        }
+
+        private const uint CRYPTPROTECTMEMORY_SAME_PROCESS = 0x00;
+
+        private static unsafe bool CryptProtectData(SafeBSTRHandle decryptedBuffer, out SafeBSTRHandle encryptedBuffer)
+        {
+            byte* uncryptedBufferPtr = null;
+            Interop.Crypt32.DATA_BLOB pDataOut = default(Interop.Crypt32.DATA_BLOB);
+            try
+            {
+                decryptedBuffer.AcquirePointer(ref uncryptedBufferPtr);
+
+                Interop.Crypt32.DATA_BLOB optionalEntropyBlob = default(Interop.Crypt32.DATA_BLOB);
+                Interop.Crypt32.DATA_BLOB pDataIn = new Interop.Crypt32.DATA_BLOB((IntPtr)uncryptedBufferPtr, decryptedBuffer.Length * 2);
+                if (Interop.Crypt32.CryptProtectData(ref pDataIn, String.Empty, ref optionalEntropyBlob, IntPtr.Zero, IntPtr.Zero, CRYPTPROTECTMEMORY_SAME_PROCESS, out pDataOut))
```

Comment:
This was all preexisting; changes I made to this file were purely minor code cleanup here and there.  I'd like to leave this as-is, and we can open an issue to follow-up on this issue separately.  We should also double-check the ProtectedData library code.  I did remove the named const since that was an easy change.